### PR TITLE
Check address sanitizer in github action

### PR DESCRIPTION
**Pull request by yixinglu**
1. Add location info of feature file when some test cases fails
2. Fix bug about load space description
3. Split TCK and pytest steps
4. Update build type with sanitizer

### CODE REVIEW

**Review by jievince — DISMISSED** [review]

**Review by laura-ding — DISMISSED** [review]

File: tests/Makefile [hunk 1]
```diff
@@ -12,6 +12,12 @@ gherkin_fmt = ~/.local/bin/reformat-gherkin
 RM_DIR ?= true
 TEST_DIR ?= $(CURR_DIR)
 
+ifeq ($(ASAN), false)
+	NUM_THREADS=8
```
Comment (Shylock-Hg) [inline tests/Makefile:16]: The option named `J` is more clear.
And `J ?= 8`.

File: tests/Makefile [hunk 2]
```diff
@@ -39,7 +45,7 @@ check:
 	@find $(CURR_DIR)/tck/features -type f -iname "*.feature" -print | xargs $(gherkin_fmt) --check
 
 test:
-	cd $(CURR_DIR) && python3 -m pytest -n8 --dist=loadfile --rm_dir=$(RM_DIR) -m "not skip" $(TEST_DIR)
+	cd $(CURR_DIR) && python3 -m pytest -n$(NUM_THREADS) --dist=loadfile --rm_dir=$(RM_DIR) -m "not skip" $(TEST_DIR)
 
 tck:
 	cd $(CURR_DIR) && python3 -m pytest --gherkin-terminal-reporter --gherkin-terminal-reporter-expanded --rm_dir=$(RM_DIR) -m "not skip" $(CURR_DIR)/tck/
```
Comment (Shylock-Hg) [inline tests/Makefile:51]: Add the job number option

**Review by CPWstatic — DISMISSED** [review]

**Review by laura-ding — DISMISSED** [review]

File: tests/Makefile (hunk 2 above)
Comment (yixinglu) [reply tests/Makefile:51]: erkin-terminal-reporter option is not compatible with xdist options

File: tests/Makefile (hunk 1 above)
Comment (yixinglu) [reply tests/Makefile]: fixed

**Review by nevermore3 — APPROVED** [review]

**Comment by bright-starry-sky** [comment]
LGTM，good job.

**Comment by yixinglu** [comment]
> LGTM，good job.

Please approve.

**Review by bright-starry-sky — APPROVED** [review]

**Pull request merged**